src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedOrderApprovedLetterService.java

```diff
@@ -54,6 +58,12 @@ public void generateAndStoreContestedOrderApprovedLetter(FinremCaseDetails finre
      */
     public void generateAndStoreContestedOrderApprovedLetter(FinremCaseDetails finremCaseDetails, String judgeDetails, String authorisationToken) {
         CaseDetails caseDetails = mapper.mapToCaseDetails(finremCaseDetails);
+        caseDetails.getData().put(CONTESTED_ORDER_APPROVED_DATE,
+            ofNullable(finremCaseDetails.getData().getDraftOrdersWrapper())
+                .map(DraftOrdersWrapper::getExtraReportFieldsInput)
+                .map(ExtraReportFieldsInput::getOrderApprovedDate)
+                .map(date -> date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd")))
+                .orElse(null));
```

Comment:
Hi @dawudgovuk.  Looks like this should work fine.

But any reason you didn't set the data on the finremCaseDetails instance passed into the method?

It just means one less place where we use CaseDetails.  You could map to CaseDetails afterward.

Reply:
Also it means the value of contested order approved date doesn't get permanently written to the case data. Does this mean it is only required for the document generation?